Modflared hash does not match

**Describe the bug**
Modflared on Macbook encounters an error again - expected hash does not match downloaded.

**Which version of the mod/game are you using?**
1.5.2 version of Modflared on 1.21.9 fabric loader version of minecraft

**To Reproduce**
Steps to reproduce the behavior:
1. Start game
2. Click on 'Multiplayer'
3. Not able to connect to hosted server 

**Expected behavior**
Modflared loading properly and tunneling connection

**Logs**
[08:58:23] [main/INFO]: Loading Minecraft 1.21.9 with Fabric Loader 0.17.3
[08:58:23] [main/INFO]: Loading 68 mods:
	- fabric-api 0.134.0+1.21.9
	   |-- fabric-api-base 1.0.0+14b92d897d
	   |-- fabric-api-lookup-api-v1 1.6.106+d17682157d
	   |-- fabric-biome-api-v1 16.1.0+c2417d4f7d
	   |-- fabric-block-api-v1 1.1.5+d17682157d
	   |-- fabric-block-view-api-v2 1.0.34+e4fd1b927d
	   |-- fabric-command-api-v2 2.3.7+328a75ba7d
	   |-- fabric-content-registries-v0 10.2.3+b6a242647d
	   |-- fabric-convention-tags-v1 2.1.47+7f945d5b7d
	   |-- fabric-convention-tags-v2 2.15.12+3154c1c87d
	   |-- fabric-crash-report-info-v1 0.3.18+33df5e6e7d
	   |-- fabric-data-attachment-api-v1 1.8.28+9bce411b7d
	   |-- fabric-data-generation-api-v1 23.2.22+d8b85ef17d
	   |-- fabric-dimensions-v1 4.0.22+38b9cc197d
	   |-- fabric-entity-events-v1 2.1.12+14b92d897d
	   |-- fabric-events-interaction-v0 4.0.35+5ba217127d
	   |-- fabric-game-rule-api-v1 1.1.1+e47eaae57d
	   |-- fabric-item-api-v1 11.5.8+9bce411b7d
	   |-- fabric-item-group-api-v1 4.2.24+86b3ae577d
	   |-- fabric-key-binding-api-v1 1.1.1+b3ba9c487d
	   |-- fabric-lifecycle-events-v1 2.6.9+33df5e6e7d
	   |-- fabric-loot-api-v2 3.0.64+3f89f5a57d
	   |-- fabric-loot-api-v3 2.0.11+d17682157d
	   |-- fabric-message-api-v1 6.1.7+3154c1c87d
	   |-- fabric-model-loading-api-v1 6.0.0+3ffd6ef07d
	   |-- fabric-networking-api-v1 5.0.13+d269470c7d
	   |-- fabric-object-builder-api-v1 21.1.24+9bce411b7d
	   |-- fabric-particles-v1 4.2.2+a0ee59a47d
	   |-- fabric-recipe-api-v1 8.1.26+d17682157d
	   |-- fabric-registry-sync-v0 6.1.41+679c8c5c7d
	   |-- fabric-renderer-indigo 4.0.19+328a75ba7d
	   |-- fabric-rendering-fluids-v1 3.1.38+86b3ae577d
	   |-- fabric-resource-conditions-api-v1 5.0.28+33df5e6e7d
	   |-- fabric-screen-api-v1 3.1.0+9c82a7987d
	   |-- fabric-screen-handler-api-v1 1.3.150+d17682157d
	   |-- fabric-serialization-api-v1 1.0.0+d857c9767d
	   |-- fabric-sound-api-v1 1.0.45+328a75ba7d
	   |-- fabric-tag-api-v1 1.2.10+2fa446b27d
	   |-- fabric-transfer-api-v1 6.0.15+79ec2c127d
	   \-- fabric-transitive-access-wideners-v1 6.5.3+41eb43437d
	- fabric-language-kotlin 1.13.7+kotlin.2.2.21
	   |-- org_jetbrains_kotlin_kotlin-reflect 2.2.21
	   |-- org_jetbrains_kotlin_kotlin-stdlib 2.2.21
	   |-- org_jetbrains_kotlin_kotlin-stdlib-jdk7 2.2.21
	   |-- org_jetbrains_kotlin_kotlin-stdlib-jdk8 2.2.21
	   |-- org_jetbrains_kotlinx_atomicfu-jvm 0.29.0
	   |-- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.10.2
	   |-- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.10.2
	   |-- org_jetbrains_kotlinx_kotlinx-datetime-jvm 0.7.1
	   |-- org_jetbrains_kotlinx_kotlinx-io-bytestring-jvm 0.8.0
	   |-- org_jetbrains_kotlinx_kotlinx-io-core-jvm 0.8.0
	   |-- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.9.0
	   |-- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.9.0
	   \-- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.9.0
	- fabricloader 0.17.3
	   \-- mixinextras 0.5.0
	- iris 1.9.6+mc1.21.9
	   |-- io_github_douira_glsl-transformer 3.0.0-pre3
	   |-- org_anarres_jcpp 1.4.14
	   \-- org_antlr_antlr4-runtime 4.13.1
	- java 21
	- minecraft 1.21.9
	- modflared 1.5.2+release.112
	- sodium 0.7.2+mc1.21.10
	   |-- fabric-renderer-api-v1 7.1.2+328a75ba6f
	   |-- fabric-rendering-v1 16.0.2+91152d886f
	   |-- fabric-resource-loader-v0 3.2.3+3388a3b26f
	   \-- fabric-resource-loader-v1 1.0.3+8cc86c156f
[08:58:23] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.7 Source=file:/Users/villhelma/Library/Application%20Support/minecraft/libraries/net/fabricmc/sponge-mixin/0.16.5+mixin.0.8.7/sponge-mixin-0.16.5+mixin.0.8.7.jar Service=Knot/Fabric Env=CLIENT
[08:58:23] [main/INFO]: Compatibility level set to JAVA_21
[08:58:23] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.json could not be read. If this is a development environment you can ignore this message
[08:58:23] [main/WARN]: Reference map 'iris-fabric.refmap.json' for mixins.iris.fabric.json could not be read. If this is a development environment you can ignore this message
[08:58:23] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.fantastic.json could not be read. If this is a development environment you can ignore this message
[08:58:23] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.vertexformat.json could not be read. If this is a development environment you can ignore this message
[08:58:23] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.bettermipmaps.json could not be read. If this is a development environment you can ignore this message
[08:58:23] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.compat.sodium.json could not be read. If this is a development environment you can ignore this message
[08:58:23] [main/WARN]: Reference map 'iris.refmap.json' for mixins.iris.fixes.maxfpscrash.json could not be read. If this is a development environment you can ignore this message
[08:58:23] [main/INFO]: Loaded configuration file for Sodium: 42 options available, 1 override(s) found
[08:58:24] [main/WARN]: Force-disabling mixin 'features.render.world.sky.FogRendererMixin' as rule 'mixin.features.render.world.sky' (added by mods [iris]) disables it and children
[08:58:24] [main/WARN]: Force-disabling mixin 'features.render.world.sky.ClientLevelMixin' as rule 'mixin.features.render.world.sky' (added by mods [iris]) disables it and children
[08:58:24] [main/WARN]: Force-disabling mixin 'features.render.world.sky.LevelRendererMixin' as rule 'mixin.features.render.world.sky' (added by mods [iris]) disables it and children
[08:58:24] [main/INFO]: Searching for graphics cards...
[08:58:24] [main/INFO]: Initializing MixinExtras via com.llamalad7.mixinextras.service.MixinExtrasServiceImpl(version=0.5.0).
[08:58:24] [Datafixer Bootstrap/INFO]: 278 Datafixer optimizations took 261 milliseconds
[08:58:28] [Render thread/INFO]: Environment: Environment[sessionHost=https://sessionserver.mojang.com, servicesHost=https://api.minecraftservices.com, profilesHost=https://api.mojang.com, name=PROD]
[08:58:28] [Render thread/INFO]: Setting user: Villhelmaa
[08:58:28] [Render thread/INFO]: [Indigo] Different rendering plugin detected; not applying Indigo.
[08:58:28] [Render thread/INFO]: Cloudflared is not installed on the system. Downloading it if necessary...
[08:58:28] [Render thread/INFO]: Backend library: LWJGL version 3.3.3-snapshot
[08:58:28] [pool-4-thread-1/INFO]: Update detected updating...
[08:58:28] [ForkJoinPool.commonPool-worker-1/WARN]: [Iris Update Check] This version doesn't have an update index, skipping.
[08:58:28] [pool-4-thread-1/INFO]: Downloading cloudflared version 2025.11.1 from github. Attempt: 1
[08:58:28] [Render thread/INFO]: Debug functionality is disabled.
[08:58:29] [Render thread/INFO]: DSA support not detected.
[08:58:29] [Render thread/INFO]: Shaders are disabled because no valid shaderpack is selected
[08:58:29] [Render thread/INFO]: OpenGL Vendor: Apple
[08:58:29] [Render thread/INFO]: OpenGL Renderer: Apple M2
[08:58:29] [Render thread/INFO]: OpenGL Version: 4.1 Metal - 90.5
[08:58:29] [Render thread/INFO]: Using optional rendering extensions: 
[08:58:29] [Render thread/INFO]: Found non-pack entry '/Users/villhelma/Library/Application Support/minecraft/resourcepacks/.DS_Store', ignoring
[08:58:29] [Render thread/INFO]: Hardware information:
[08:58:29] [Render thread/INFO]: CPU: 8x Apple M2
[08:58:29] [Render thread/INFO]: GPU: Apple M2 (Supports OpenGL 4.1 Metal - 90.5)
[08:58:29] [Render thread/INFO]: OS: Mac OS X (26.0.1)
[08:58:30] [Render thread/INFO]: Reloading ResourceManager: vanilla, file/Mizunos 16 Craft JE_1.20.1-1.0_230720, fabric, fabric-api, fabric-api-base, fabric-api-lookup-api-v1, fabric-biome-api-v1, fabric-block-api-v1, fabric-block-view-api-v2, fabric-command-api-v2, fabric-content-registries-v0, fabric-convention-tags-v1, fabric-convention-tags-v2, fabric-crash-report-info-v1, fabric-data-attachment-api-v1, fabric-data-generation-api-v1, fabric-dimensions-v1, fabric-entity-events-v1, fabric-events-interaction-v0, fabric-game-rule-api-v1, fabric-item-api-v1, fabric-item-group-api-v1, fabric-key-binding-api-v1, fabric-language-kotlin, fabric-lifecycle-events-v1, fabric-loot-api-v2, fabric-loot-api-v3, fabric-message-api-v1, fabric-model-loading-api-v1, fabric-networking-api-v1, fabric-object-builder-api-v1, fabric-particles-v1, fabric-recipe-api-v1, fabric-registry-sync-v0, fabric-renderer-api-v1, fabric-renderer-indigo, fabric-rendering-fluids-v1, fabric-rendering-v1, fabric-resource-conditions-api-v1, fabric-resource-loader-v0, fabric-resource-loader-v1, fabric-screen-api-v1, fabric-screen-handler-api-v1, fabric-serialization-api-v1, fabric-sound-api-v1, fabric-tag-api-v1, fabric-transfer-api-v1, fabric-transitive-access-wideners-v1, fabricloader, iris, modflared, sodium, file/VanillaTweaks_r958396_MC1.21.x, file/autumn_birch_1.21.7
[08:58:30] [Worker-Main-5/INFO]: Found unifont_pua-16.0.03.hex, loading
[08:58:30] [Worker-Main-7/INFO]: Found unifont_all_no_pua-16.0.03.hex, loading
[08:58:30] [Worker-Main-1/INFO]: Found unifont_jp_patch-16.0.03.hex, loading
[08:58:30] [Worker-Main-4/WARN]: Unused frames in sprite minecraft:entity/conduit/wind: [22, 23, 24, 25, 26, 27, 28, 29, 30, 31]
[08:58:30] [Worker-Main-4/WARN]: Unused frames in sprite minecraft:entity/conduit/wind_vertical: [22, 23, 24, 25, 26, 27, 28, 29, 30, 31]
[08:58:31] [Worker-Main-2/WARN]: Missing block model: minecraft:item/copper_spear
[08:58:31] [Worker-Main-2/WARN]: Missing block model: minecraft:item/copper_spear_in_hand
[08:58:31] [Worker-Main-2/WARN]: Missing block model: minecraft:item/spear_in_hand
[08:58:31] [Worker-Main-1/WARN]: Unresolved texture references in minecraft:block/sandstone_wall_inventory:
	#particle-> #wall

[08:58:31] [Worker-Main-1/WARN]: Unresolved texture references in minecraft:block/red_sandstone_wall_inventory:
	#particle-> #wall

[08:58:31] [Worker-Main-6/WARN]: Missing textures in model minecraft:block/cake_slice3_0:
    minecraft:textures/atlas/blocks.png:minecraft:blocks/cake_side_0
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:item/cherry_leaves:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:item/jungle_leaves:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:item/dark_oak_leaves:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:block/red_sandstone_wall_inventory:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:item/azalea_leaves:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:item/spruce_leaves:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:item/birch_leaves:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:item/flowering_azalea_leaves:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:block/sandstone_wall_inventory:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:item/oak_leaves:
    particle
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:block/piston_head_short_sticky:
    #arm
[08:58:31] [Worker-Main-6/WARN]: Missing texture references in model minecraft:item/acacia_leaves:
    particle
[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_text_background_see_through and FS minecraft:core/rendertype_text_background_see_through. Log output: WARNING: Could not find vertex shader attribute 'UV2' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/entity and FS minecraft:core/entity. Log output: WARNING: Could not find vertex shader attribute 'UV2' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/entity and FS minecraft:core/entity. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.
WARNING: Could not find vertex shader attribute 'UV2' to match BindAttributeLocation request.
WARNING: Output of vertex shader 'overlayColor' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/terrain and FS minecraft:core/terrain. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_beacon_beam and FS minecraft:core/rendertype_beacon_beam. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.
WARNING: Could not find vertex shader attribute 'UV2' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_translucent_moving_block and FS minecraft:core/rendertype_translucent_moving_block. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/terrain and FS minecraft:core/terrain. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_entity_alpha and FS minecraft:core/rendertype_entity_alpha. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.
WARNING: Output of vertex shader 'texCoord1' not read by fragment shader
WARNING: Output of vertex shader 'texCoord2' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/entity and FS minecraft:core/entity. Log output: WARNING: Output of vertex shader 'overlayColor' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/terrain and FS minecraft:core/terrain. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/entity and FS minecraft:core/entity. Log output: WARNING: Output of vertex shader 'overlayColor' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/entity and FS minecraft:core/entity. Log output: WARNING: Output of vertex shader 'overlayColor' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_crumbling and FS minecraft:core/rendertype_crumbling. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.
WARNING: Output of vertex shader 'texCoord2' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/entity and FS minecraft:core/entity. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.
WARNING: Output of vertex shader 'overlayColor' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/entity and FS minecraft:core/entity. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.
WARNING: Could not find vertex shader attribute 'UV2' to match BindAttributeLocation request.
WARNING: Output of vertex shader 'overlayColor' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/terrain and FS minecraft:core/terrain. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_text_see_through and FS minecraft:core/rendertype_text_see_through. Log output: WARNING: Could not find vertex shader attribute 'UV2' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_text_intensity_see_through and FS minecraft:core/rendertype_text_intensity_see_through. Log output: WARNING: Could not find vertex shader attribute 'UV2' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_item_entity_translucent_cull and FS minecraft:core/rendertype_item_entity_translucent_cull. Log output: WARNING: Output of vertex shader 'texCoord1' not read by fragment shader
WARNING: Output of vertex shader 'texCoord2' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_beacon_beam and FS minecraft:core/rendertype_beacon_beam. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.
WARNING: Could not find vertex shader attribute 'UV2' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/rendertype_entity_shadow and FS minecraft:core/rendertype_entity_shadow. Log output: WARNING: Could not find vertex shader attribute 'UV1' to match BindAttributeLocation request.
WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.
WARNING: Could not find vertex shader attribute 'UV2' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/terrain and FS minecraft:core/terrain. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/entity and FS minecraft:core/entity. Log output: WARNING: Output of vertex shader 'overlayColor' not read by fragment shader

[08:58:31] [Render thread/INFO]: Info log when linking program containing VS minecraft:core/terrain and FS minecraft:core/terrain. Log output: WARNING: Could not find vertex shader attribute 'Normal' to match BindAttributeLocation request.

[08:58:31] [Render thread/INFO]: OpenAL initialized on device MacBook Air Speakers
[08:58:31] [Render thread/INFO]: Sound engine started
[08:58:31] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/particles.png-atlas
[08:58:31] [Render thread/INFO]: Created: 128x64x0 minecraft:textures/atlas/decorated_pot.png-atlas
[08:58:31] [Render thread/INFO]: Created: 2048x1024x0 minecraft:textures/atlas/armor_trims.png-atlas
[08:58:31] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/paintings.png-atlas
[08:58:31] [Render thread/INFO]: Created: 1024x1024x0 minecraft:textures/atlas/shield_patterns.png-atlas
[08:58:31] [Render thread/INFO]: Created: 1024x1024x4 minecraft:textures/atlas/blocks.png-atlas
[08:58:31] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/chest.png-atlas
[08:58:31] [Render thread/INFO]: Created: 512x512x0 minecraft:textures/atlas/banner_patterns.png-atlas
[08:58:31] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/beds.png-atlas
[08:58:31] [Render thread/INFO]: Created: 1024x512x0 minecraft:textures/atlas/gui.png-atlas
[08:58:31] [Render thread/INFO]: Created: 64x64x0 minecraft:textures/atlas/map_decorations.png-atlas
[08:58:31] [Render thread/INFO]: Created: 256x256x0 minecraft:textures/atlas/signs.png-atlas
[08:58:31] [Render thread/INFO]: Created: 512x256x0 minecraft:textures/atlas/shulker_boxes.png-atlas
[08:58:31] [Render thread/INFO]: Creating pipeline for dimension minecraft:overworld
[08:58:35] [Server Pinger #0/WARN]: Modflared is not ready yet, ignoring all connections.
[08:58:50] [pool-4-thread-1/INFO]: Checking file integrity
[08:58:50] [pool-4-thread-1/WARN]: This downloaded file does not match with the file hash provided on GitHub.
[08:58:50] [pool-4-thread-1/WARN]: Expected fe42668939aeaff2a7f4153dc80768a5a579f7eff5b7a241e395d01f651add7c, Provided: 45cfbb59a720f60b873906aa6469f8c4058f26be6d351c3e2920bc9cb4714273
[08:58:50] [pool-4-thread-1/INFO]: Downloading cloudflared version 2025.11.1 from github. Attempt: 2
[08:59:16] [pool-4-thread-1/INFO]: Checking file integrity
[08:59:16] [pool-4-thread-1/WARN]: This downloaded file does not match with the file hash provided on GitHub.
[08:59:16] [pool-4-thread-1/WARN]: Expected fe42668939aeaff2a7f4153dc80768a5a579f7eff5b7a241e395d01f651add7c, Provided: 45cfbb59a720f60b873906aa6469f8c4058f26be6d351c3e2920bc9cb4714273
[08:59:16] [pool-4-thread-1/INFO]: Downloading cloudflared version 2025.11.1 from github. Attempt: 3
[08:59:34] [Server Pinger #0/WARN]: Modflared is not ready yet, ignoring all connections.
[08:59:36] [pool-4-thread-1/INFO]: Checking file integrity
[08:59:36] [pool-4-thread-1/WARN]: This downloaded file does not match with the file hash provided on GitHub.
[08:59:36] [pool-4-thread-1/WARN]: Expected fe42668939aeaff2a7f4153dc80768a5a579f7eff5b7a241e395d01f651add7c, Provided: 45cfbb59a720f60b873906aa6469f8c4058f26be6d351c3e2920bc9cb4714273
[08:59:36] [pool-4-thread-1/INFO]: Downloading cloudflared version 2025.11.1 from github. Attempt: 4
[08:59:56] [pool-4-thread-1/INFO]: Checking file integrity
[08:59:56] [pool-4-thread-1/WARN]: This downloaded file does not match with the file hash provided on GitHub.
[08:59:56] [pool-4-thread-1/WARN]: Expected fe42668939aeaff2a7f4153dc80768a5a579f7eff5b7a241e395d01f651add7c, Provided: 45cfbb59a720f60b873906aa6469f8c4058f26be6d351c3e2920bc9cb4714273
[08:59:56] [pool-4-thread-1/ERROR]: Failed to download update
java.util.concurrent.CompletionException: java.lang.IllegalStateException: Modflared failed 5 times to download cloudflared from github. Please check your internet connection
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: java.lang.IllegalStateException: Modflared failed 5 times to download cloudflared from github. Please check your internet connection
	at knot/de.rafael.modflared.binary.download.DownloadedCloudflared.lambda$downloadFile$5(DownloadedCloudflared.java:149) ~[modflared-1.5.2+release.112.jar:?]
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718) ~[?:?]
	... 4 more


**Additional context**
It was already a problem a week ago. Seems to happen every cloudflare update.

--- a/common/src/main/java/de/rafael/modflared/binary/download/DownloadedCloudflared.java
+++ b/common/src/main/java/de/rafael/modflared/binary/download/DownloadedCloudflared.java
@@ -117,21 +117,22 @@ public class DownloadedCloudflared extends Cloudflared {
                 for (int i = 0; i < 4; i++) {
                     Modflared.LOGGER.info("Downloading cloudflared version {} from github. Attempt: {}", version, i + 1);
                     var downloadedFile = syncDownloadFile();
+                    Modflared.LOGGER.info("Downloaded file preparing cloudflared binary...");
                     var file = new File(TunnelManager.DATA_FOLDER, download.fileName());
+                    prepareFile(downloadedFile, file);
+
                     // Check if file is corrupt
                     Modflared.LOGGER.info("Checking file integrity");
-                    var provided = GithubAPI.FileHash.computeHash(downloadedFile);
+                    var provided = GithubAPI.FileHash.computeHash(file);
                     if(expected.compareTo(provided)) {
                         Modflared.LOGGER.info("Download finished of cloudflared version {}!", version);
-                        Modflared.LOGGER.info("Downloaded file preparing cloudflared binary...");
-                        prepareFile(downloadedFile, file);
                         return;
                     } else {
                         Modflared.LOGGER.warn("This downloaded file does not match with the file hash provided on GitHub.");
                         Modflared.LOGGER.warn("Expected {}, Provided: {}", expected.hash(), provided.hash());
-                        downloadedFile.delete();
-                    }
 
+                        file.delete();
+                    }
                 }
             } catch (InterruptedException exception) {
                 throw new IllegalStateException("Error while unpacking MacOS cloudflared download", exception);

